File: backend/backend/src/main/java/dom/lot/backend/model/Flight.java
package dom.lot.backend.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Duration;
import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Flight {
    private String flightNumber;
    private String placeOfDeparture;
    private String placeOfArrival;
    private Duration flightDuration;
    private boolean isOnWayFlight;
    private List<String> availableSeats;
}

File: frontend/frontend/src/types/Flight.ts
export interface Flight {
    flightNumber: string;
    placeOfDeparture: string;
    placeOfArrival: string;
    flightDuration: string; // ISO 8601 duration, e.g. "PT2H30M"
    isOnWayFlight: boolean;
    availableSeats: string[]; // array of seat identifiers like "1A", "1B"
}
